lib/archethic/contracts/interpreter/function_interpreter.ex

```diff
@@ -98,18 +97,78 @@ defmodule Archethic.Contracts.Interpreter.FunctionInterpreter do
   #  | .__/|_|  \___| \_/\_/ \__,_|_|_|\_\
   #  |_|
   # ----------------------------------------------------------------------
-
   # Ban access to Contract module
   defp prewalk(
-         node = {:__aliases__, _, [atom: "Contract"]},
-         _
+         node =
+           {{:., _meta, [{:__aliases__, _, [atom: "Contract"]}, {:atom, function_name}]}, _, _},
```

Comment:
I you create a `Library.module_exists?` function, you could create a single prewalk that control both io and write contract tags